moto/rds3/models/db_snapshot.py

```diff
@@ -157,5 +157,24 @@ def describe_db_snapshots(self, db_instance_identifier=None, db_snapshot_identif
                 if snapshot.db_instance_identifier == db_instance_identifier:
                     if snapshot.snapshot_type in snapshot_types:
                         db_instance_snapshots.append(snapshot)
-            return db_instance_snapshots
-        return self.db_snapshots.values()
+            if marker:
+                start = db_instance_snapshots.index(marker) + 1
```

Comment:
what if the marker is out of the bounds of the index ?

Reply:
Also, what if db_instance_snapshots[] is empty. Add a check?

Reply:
>Marker is an optional pagination token provided by a previous request. If this parameter is specified, the response includes only records beyond the marker, up to the value specified by MaxRecords. 

As quoted in this, marker generally is not set by user and used for pagination purpose.